File: tests/fixtures/edgecase_repo/lua/module1.lua
local mod0 = require("module0")

local Base1 = {}
Base1.__index = Base1

function Base1.new(name)
  local self = setmetatable({}, Base1)
  self.name = name
  return self
end

local Service1 = setmetatable({}, { __index = Base1 })

function Service1:handle(payload)
  self:validate(payload)
  return true
end

function Service1:validate(payload)
  return payload ~= nil
end

local function helper1(value)
  return value
end

return Service1

File: tests/fixtures/edgecase_repo/python/pkg/module1.py
"""Module 1 of the generated Python layer."""

from __future__ import annotations

from pkg.module0 import Service0


class Base1:
    """Base class for service 1."""


class Service1(Base1):
    """Service 1."""

    def handle(self, payload):
        """Handle a payload."""
        self.validate(payload)
        return Service0().handle(payload)

    def validate(self, payload):
        """Validate a payload."""
        return payload is not None


def helper_1(value):
    """Module-level helper."""
    return value
